### PR TITLE
add warning message for otel-provider

### PR DESCRIPTION
**Pull request by zirain**
Please provide a description for what this PR is for.

add warning message mentioned in https://github.com/istio/istio.io/pull/11058#issuecomment-1101485559

cc @ericvn 

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/observability/logs/otel-provider/index.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ owner: istio/wg-policies-and-telemetry-maintainers
 test: yes
 ---
 
+{{< warning >}}
+Use of the OpenTelemetry provider is avialiable in v1.13.3 or later.
```
Comment (craigbox) [inline content/en/docs/tasks/observability/logs/otel-provider/index.md:11]:
```suggestion
The OpenTelemetry provider is available in Istio v1.13.3 or later.
```

**Review by craigbox — CHANGES_REQUESTED** [review]

File: content/en/docs/tasks/observability/logs/otel-provider/index.md (hunk 1 above)
Comment (zirain) [reply content/en/docs/tasks/observability/logs/otel-provider/index.md]: updated

File: content/en/docs/tasks/observability/logs/otel-provider/index.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ owner: istio/wg-policies-and-telemetry-maintainers
 test: yes
 ---
 
+{{< warning >}}
+The OpenTelemetry provider is available in Istio `v1.13.3` or later.
```
Comment (ericvn) [inline content/en/docs/tasks/observability/logs/otel-provider/index.md:11]: On other parts the page and in the title, we say collector. Should this be the same?

Comment (zirain) [reply content/en/docs/tasks/observability/logs/otel-provider/index.md:11]: `OpenTelemetry provider` is a extension provider type in Istio.

**Comment by zirain** [comment]
we do not need this warning message in 1.14, so target branch is `release-1.13`, cherry-pick is not required.

**Review by craigbox — APPROVED** [review]
Approving, on the assumption that this will go away for 1.14 anyway

**Pull request merged**